### PR TITLE
Adding more tests and new problems

### PR DESCRIPTION
**Pull request by acasagran**
Adding tests for error checking and more engineering type problems using CSVDiff. Also added more output quantities and used the GeneratedMesh capability.

### CODE REVIEW

**Comment by acasagran** [comment]
Swetha, here's some more changes related to error checking and additional test problems. Let me know if you have any questions. We can work on this more next week too.

**Comment by sveerara** [comment]
Al, thanks for adding the error checks and editing/adding more tests. I will go through the tests later today.

File: modules/tensor_mechanics/test/tests/beam/action/tests [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
-    test = Exodiff
+    type = 'Exodiff'
     input = '2_block_common.i'
     exodiff = '2_block_out.e'
     prereq = '2_block_action'
   [../]
+  [./beam_action_test1]
+    type = 'RunException'
+    input = 'beam_action_chk.i'
+    expect_err = 'LineElementAction: A vector of displacement variable names should be provided as'
```
Comment (sveerara) [inline modules/tensor_mechanics/test/tests/beam/action/tests:16]: Shouldn't the full error statement be provided here? The last part of this statement is 'input using `displacements`.'

File: modules/tensor_mechanics/test/tests/beam/action/tests [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+  [../]
+  [./beam_action_test2]
+    type = 'RunException'
+    input = 'beam_action_chk.i'
+    cli_args = 'Modules/LineElement/block_1/displacements=""'
+    expect_err = 'LineElementAction: Rotational variable names should be provided for beam elements using `rotations` parameter'
```
Comment (sveerara) [inline modules/tensor_mechanics/test/tests/beam/action/tests:22]: full stop at end of parameter

File: modules/tensor_mechanics/test/tests/beam/dynamic/dyn_euler_small_added_mass2.i [hunk 3]
```diff
@@ -0,0 +1,347 @@
+# Test for small strain euler beam vibration in y direction
+
+# An impulse load is applied at the end of a cantilever beam of length 5ft (60 in).
+# The beam is massless with a lumped mass at the end of the beam of 5000 lb
+# The properties of the cantilever beam are as follows:
+# E = 1e7 and I = 120 in^4
+# Assuming a square cross section A = sqrt(12 * I) = 37.95
+# Shear modulus (G) = 3.846e6
+# Shear coefficient (k) = 1.0
+# Cross-section area (A) = 1.0
```
Comment (sveerara) [inline modules/tensor_mechanics/test/tests/beam/dynamic/dyn_euler_small_added_mass2.i:10]: There is another line above with cross-section area

File: modules/tensor_mechanics/test/tests/beam/eigenstrain/tests [hunk 4]
```diff
@@ -20,4 +20,22 @@
...
+    type = 'RunException'
+    input = 'eigenstrain_from_csv.i'
+    cli_args = 'UserObjects/disp_uo/variable_vectors=""'
+    expect_err = 'CSVInterpolator: Please provide a non-empty list of variable vectors.'
+  [../]
+  [./eigenstrain_from_csv_test1]
+    type = 'RunException'
+    input = 'eigenstrain_from_csv.i'
+    cli_args = 'Materials/thermal/to_component=3'
+    expect_err = 'ComputeEigenstrainBeamFromCSVInterpolator: to_component should be between 0 and 2'
```
Comment (sveerara) [inline modules/tensor_mechanics/test/tests/beam/eigenstrain/tests:33]: full stop after 2

**Review by sveerara — CHANGES_REQUESTED** [review]
There are a few comments with respect to the error messages. Other than that, the tests look great and are very useful. I din't realize that I dint have any axial tests apart from the thermal expansion test. Thanks for working on this, Al.

File: modules/tensor_mechanics/test/tests/beam/action/tests (hunk 1 above)
Comment (acasagran) [reply modules/tensor_mechanics/test/tests/beam/action/tests:16]: I think the expect_err just has to be unique, so you don't have to put the whole thing in there, but if you want to for completeness, go ahead.

File: modules/tensor_mechanics/test/tests/beam/dynamic/dyn_euler_small_added_mass2.i (hunk 3 above)
Comment (acasagran) [reply modules/tensor_mechanics/test/tests/beam/dynamic/dyn_euler_small_added_mass2.i:10]: I started off with the cross section being 1.0, but realized I needed to calculate it from the value of I given in this problem. I would just delete the line with A = 1.0.

File: modules/tensor_mechanics/test/tests/beam/action/tests (hunk 1 above)
Comment (sveerara) [reply modules/tensor_mechanics/test/tests/beam/action/tests:16]: Thanks, Al. I did not know that.

File: modules/tensor_mechanics/test/tests/beam/dynamic/dyn_euler_small_added_mass2.i (hunk 3 above)
Comment (sveerara) [reply modules/tensor_mechanics/test/tests/beam/dynamic/dyn_euler_small_added_mass2.i:10]: I will approve the merge request and then make this change along with the updates to the documentation.

**Review by sveerara — APPROVED** [review]

**Pull request merged**